### ISSUE TITLE
[New Package]: Tencent.TencentMeeting

### ISSUE THREAD

**Issue by SpecterShell** [vedantmgoyal2009/vedantmgoyal2009#230]
### List the PackageIdentifier of the package to be added to automation

Tencent.TencentMeeting

### Details about the application

```powershell
irm 'https://meeting.tencent.com/wemeet-webapi/v2/config/query-download-info' -Method POST -Body "[{`"instance`":`"windows`",`"type`":`"0300000000`"}]"
```

PackageVersion: `$result.data[0].version`
InstallerUrl: `$result.data[0].url`

**Comment by vedantmgoyal2009** [comment]
The source is still displaying the previous version (3.3.4.444) and an updated version (3.4.0.423) is already there in the repository here: https://github.com/microsoft/winget-pkgs/tree/master/manifests/t/Tencent/TencentMeeting.

**Comment by SpecterShell** [comment]
The current version is a beta version. 🤔

**Comment by lychichem** [comment]
> The source is still displaying the previous version (3.3.4.444) and an updated version (3.4.0.423) is already there in the repository here: https://github.com/microsoft/winget-pkgs/tree/master/manifests/t/Tencent/TencentMeeting.

This is beta version. However, the problem is, even if you install stable version on your machine, every time you start the software, tencent will check for beta version.

**Comment by vedantmgoyal2009** [comment]
Strange, so which source should I add, you guys only tell since I don't know about the software. My opinion would be to add beta version.

**Comment by lychichem** [comment]
> Strange, so which source should I add, you guys only tell since I don't know about the software. My opinion would be to add beta version.

I made a test, here I'll tell you my result:

1. Download stable version of tencentmeeting from [link](https://meeting.tencent.com/download-win.html?from=1000&fromSource=1) given by @SpecterShell 
![屏幕截图 2022-01-15 162056](https://user-images.githubusercontent.com/9316590/149615887-92d40203-748e-432c-b8af-d4dcaf25735b.png)
2. Install it by using command below
![屏幕截图 2022-01-15 162143](https://user-images.githubusercontent.com/9316590/149615886-5b37d46c-04a7-4b70-b688-32aecc40d454.png)
3. Open it and check for update, it shows new version is avaliable
![屏幕截图 2022-01-15 162648](https://user-images.githubusercontent.com/9316590/149615884-8d9035c6-f019-4295-bf66-0df249121409.png)
4. Install the new version, it's neither 3.3.4.444 nor beta version 3.4.1.436 but 3.3.5.419
![屏幕截图 2022-01-15 162819](https://user-images.githubusercontent.com/9316590/149615883-c2f018b3-765b-4e61-80d0-c82520741f3c.png)
5. Directly install tencentmeeting by running `winget install tencent.tencentmeeting` in shell with admin previliedge, then open it, it will say "no new version detected"

**Comment by lychichem** [comment]
I guess the issue will happen when the version number of stable version is the same as beta version, the program cannot distinguish which channel it belongs to.
Another problem is, this program can only be installed silently when you execuate the installer in admin shell. If not, nothing will be really installed, and the installer will delete itself and return northing. And, even if successfully installed, the program will not write `uninstallstring` in registry, which means you cannot uninstall it by running `winget uninstall`, and when updating from former version, te old version will not be removed. But if you run the uninstaller in one version folder, all versions of this program will be removed. 

So I only want to say: F♂q tencent👎

**Comment by vedantmgoyal2009** [comment]
The software developers at Tencent don't know about release management 🤷‍♂️ I guess I should beta version only then because if Tencent doesn't care then neither WinGet and nor me should care about and let the user face the inconvenience and the company should fix it.

**Comment by SpecterShell** [comment]
![image](https://user-images.githubusercontent.com/56779163/149730807-424d944e-9a51-402c-880d-615c97427ab9.png)
It is strange that it didn't push any updates for me here.

**Comment by SpecterShell** [comment]
If this situation comes from some A/B tests, then I prefer to select release channel for this automation and let users to upgrade themseleves.🤷
By the way, this software uses delta updates, so there is no use to capture the upgrade API.

**Comment by lychichem** [comment]
> ![image](https://user-images.githubusercontent.com/56779163/149730807-424d944e-9a51-402c-880d-615c97427ab9.png)
> It is strange that it didn't push any updates for me here.

I noticed you are using English for Tencent meeting. But I tested with using Chinese. Is this the reason?

**Comment by SpecterShell** [comment]
![image](https://user-images.githubusercontent.com/56779163/149776711-ede98fea-da68-4878-ae8f-1a9836d7ef6c.png)

**Comment by lychichem** [comment]
> ![image](https://user-images.githubusercontent.com/56779163/149776711-ede98fea-da68-4878-ae8f-1a9836d7ef6c.png)

ok, maybe stable version will be better, as Tencent release a new beta version nearly every week.

**Comment by lychichem** [comment]
@SpecterShell @vedantmgoyal2009 As tencent just released 3.4.2.429 as stable version on Jan 18, and beta version is also on 3.4.2.429 now, should we test both channels again?

**Comment by SpecterShell** [comment]
Even the download links of both stable (3.4.3.477) and beta (3.4.3.477) versions are the same. Can the software itself distinguish which channel it belongs?

**Comment by lychichem** [comment]
> Even the download links of both stable (3.4.3.477) and beta (3.4.3.477) versions are the same. Can the software itself distinguish which channel it belongs?

Maybe we need more tests.

**Comment by vedantmgoyal9** [comment]
Should we go with beta channel or _more tests_?

**Comment by SpecterShell** [comment]
Stable version should be enough.

**Comment by lychichem** [comment]
> Should we go with beta channel or _more tests_?

Now stable version will not suggest beta versions anymore. Stable is enough.

**Comment by vedantmgoyal9** [comment]
> The software developers at Tencent don't know about release management 🤷‍♂️ I guess I should beta version only then because if Tencent doesn't care then neither WinGet and nor me should care about and let the user face the inconvenience and the company should fix it.

Looks like someone from Tencent read my comment and fixed the issue 😂

**Comment by SpecterShell** [comment]
https://meeting.tencent.com/web-service/query-download-info?q=[{%22package-type%22:%22app%22,%22channel%22:%220300000000%22,%22platform%22:%22windows%22}]&nonce=AAAAAAAAAAAAAAAA

PackageVersion: `$Response.'info-list'[0].version`
InstallerUrl: `$Response.'info-list'[0].url`
ReleaseDate: `$Response.'info-list'[0].'sub-date'`

**Comment by SpecterShell** [comment]
The update API, which takes me too much effort to obtain. 🤯
`https://meeting.tencent.com/web-service/query-app-update-info/?os=Windows&sdk_id=0300000000&from=2&appver=${LastVersion}`
Now the package should always be the latest.

**Comment by vedantmgoyal2009** [comment]
Resolved in commit 86e644b66f61c907028aba1dc6b1f466a73af91d.

**Comment by vedantmgoyal9** [comment]
Resolved in commit 86e644b66f61c907028aba1dc6b1f466a73af91d.
